### DISCUSSION TITLE
Migration to independent platform for provision of source code

### DISCUSSION THREAD

**Discussion by linux-lukas** [uxbox/uxbox#2106] [General]
### A. Problem / Goal
Since the purchase of GitHub by Microsoft in 2018, a dependence on the BigTech corporation can no longer be denied.

On the one hand, I can understand why GitHub was chosen as the platform for making source code available: "Everyone is here".

On the other hand, I see the danger of a vendor-lockin effect and that open source projects become centrally dependent on Microsoft. In my eyes, this is very dangerous for free and open source software and hardware projects.

I am aware of the fact that such a migration process requires more personnel and possibly also financial resources and cannot be carried out at once, but rather step by step. 

In the medium and long term, the goal would be to become independent of GitHub and thus of Microsoft. The Gitea-based Codeberg project of Codeberg e. V. in Berlin would be a good choice here.

There are also (legal) problems with compliance with the licence of GitHub functions, such as the co-pilot.

### B. Solution
My considered solution to the problem described in A. would be the following:

1. A user of this open source project creates a user account on https://codeberg.org/
If necessary: This user creates an organisation for the project.
2. A "personal access token" is created on the GitHub account, which has appropriate rights to the organisation repositories, using the developer options in the settings.
3. All repositories would be migrated with this access token into the ownership of the organisation created in step two.
4. Regarding step four, there is an entry in the documentation of Codeberg: https://docs.codeberg.org/advanced/migrating-repos/

### C. Alternatives
A possible alternative would be to perform the first three steps as described in B. A possible alternative would be to perform the first three steps as described in B., and modify the fourth step to include a mirror of GitHub. So that all issues and such that would be created in the GitHub repository would be transferred to the Codeberg repository.

### D. Responsibilities
I would see the responsibility in the owners of the repository and, if necessary, additional project participants.

### E. Other
Basically, a look at the documentation of Codeberg is not unwise: https://docs.codeberg.org/

Should it be necessary to manage repositories in organisations, this is also possible under Codeberg, see: https://docs.codeberg.org/collaborating/create-organization/

Regarding licensing there is a page in the documentation of Codeberg: https://docs.codeberg.org/getting-started/licensing/

Funfact 1: The Pixelfed maintainer has written on Mastodon that he is considering migrating the Pixelfed project away from GitHub, to Codeberg: https://mastodon.social/@dansup/10869007146309141

Funfact 2: Aral had migrated - also on Mastodon - to Codeberg for his GitHub projects as well as those of his Foundation, which he organised together with another person, Small Technology, or is still migrating: https://mastodon.ar.al/@aral/108651118032076133

**Comment by airtonix** [comment]
The reason why people choose GitHub had nothing to do with "people are here"and everything to do with features and experience of collaboration.

GitHub, it's more than mere git hosting.

None of the others provide the quality experience that GitHub does.

**Comment by niwinz** [comment]
I think this is a better fit on: https://community.penpot.app/, github discussions can be considered legacy.

In any case, we have no hard lock on github and the issues on github are just informative for people who reports them, we migrate all the tickets to our internal [taiga](https://taiga.io) project. So we use github just for make it easily findable or in other words because "everyone is there".

Personally, if I migrate away of github, I probably will use https://sourcehut.org/
